### PR TITLE
Yeet CASC (WIP)

### PR DESCRIPTION
**Pull request by netadr**
This is a work-in-progress effort to provide the ability to change the active implementation of `DB_FileSysInterface` to `DiskFS` instead of `BnetTACTVFSManager` (default). Using `DiskFS` as the active implementation allows for game files to be loaded directly from the filesystem, bypassing battle.net's CASC containers entirely.

### CODE REVIEW

File: src/client/component/database.cpp [hunk 1]
```diff
@@ -39,7 +39,7 @@ namespace database
 				nullptr
 			};
 
-			int default_value = utils::io::directory_exists("Data/data") ? 0 : 1;
+			int default_value = (utils::io::directory_exists("Data/data") && utils::io::directory_exists("Data/config") && utils::io::directory_exists("Data/indices")) ? 0 : 1;
```
Comment (fedddddd) [inline src/client/component/database.cpp:42]: const auto default_value = static_cast<int>(...

Comment (fedddddd) [inline src/client/component/database.cpp:42]: const auto default_value = static_cast<int>(...

File: src/client/component/database.cpp [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+		
+	}
+
+	class component final : public component_interface
+	{
+	public:
+		void post_unpack() override
+		{
+			static const char* values[] = {
```
Comment (fedddddd) [inline src/client/component/database.cpp:36]: newline

File: src/client/component/database.cpp [hunk 3]
```diff
@@ -4,18 +4,17 @@
...
-#include <utils/flags.hpp>
+#include <utils/io.hpp>
 #include <utils/hook.hpp>
 
 namespace database
 {
 	namespace
 	{
 		game::dvar_t* db_filesysImpl = nullptr;
+		utils::hook::detour db_fsinitialize_hook;
```
Comment (fedddddd) [inline src/client/component/database.cpp:15]: db_fs_initialize_...

File: src/client/component/database.cpp (hunk 1 above)
Comment (fedddddd) [reply src/client/component/database.cpp:42]: also put a newline in between one of the &&

**Review by fedddddd — CHANGES_REQUESTED** [review]

File: src/client/component/database.cpp [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+		void post_unpack() override
+		{
+			static const char* values[] = 
+			{
+				"BnetTACTVFSManager", // (load files from CASC)
+				"DiskFS", // (load files from disk)
+				nullptr
+			};
+
+			const auto default_value = static_cast<int>(!utils::io::directory_exists("Data/data")
```
Comment (diamante0018) [inline src/client/component/database.cpp:43]: This static cast is not necessary. Bool will be promoted to an int when call register enum.
https://docs.microsoft.com/en-us/cpp/cpp/type-conversions-and-type-safety-modern-cpp?view=msvc-170#widening-conversions-promotion
Just noticed this now 😅

**Pull request merged**